DOCSUP-9860: document refactored storage dictionary

lehasm opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation for #23436

lehasm committed:
typo

lehasm committed:
Merge branch 'master' of https://github.com/ClickHouse/ClickHouse into alexey-sm-DOCSUP-9860-document-refactored-storage-dictionary

lehasm committed:
First draft

lehasm committed:
SET moved in the TOC

lehasm committed:
Updates

lehasm commented:
Hello.
Noticed that exception messages for dictionary queries contain _table_ , not _dictionary_ . Although it is not directly related to the documentation.
```DETACH DICTIONARY alexey_sm.d0 PERMANENTLY;
Received exception from server (version 21.9.1):
Code: 60. DB::Exception: Received from localhost:9000. DB::Exception: Table `alexey_sm`.`d0` doesn't exist. (UNKNOWN_TABLE)
```

olgarev commented on docs/en/sql-reference/statements/detach.md:
```diff
@@ -5,17 +5,18 @@ toc_title: DETACH
 # DETACH Statement {#detach}
 
-Makes the server "forget" about the existence of the table or materialized view.
+Makes the server "forget" about the existence of a table, a materialized view or a dictionary.
```
```suggestion
Makes the server "forget" about the existence of a table, a materialized view, or a dictionary.
```

olgarev commented on docs/en/sql-reference/statements/detach.md:
````diff
@@ -5,17 +5,18 @@ toc_title: DETACH
 ```
 
-Detaching does not delete the data or metadata for the table or materialized view. If the table or view was not detached `PERMANENTLY`, on the next server launch the server will read the metadata and recall the table/view again. If the table or view was detached `PERMANENTLY`, there will be no automatic recall.
+Detaching does not delete the data or metadata of the table, the materialized view or the dictionary. If the entity was not detached `PERMANENTLY`, on the next server launch the server will read the metadata and recall the table/view/dictionary again. If the entity was detached `PERMANENTLY`, there will be no automatic recall.
````
```suggestion
Detaching does not delete the data or metadata of a table, a materialized view or a dictionary. If an entity was not detached `PERMANENTLY`, on the next server launch the server will read the metadata and recall the table/view/dictionary again. If an entity was detached `PERMANENTLY`, there will be no automatic recall.
```

olgarev commented on docs/en/sql-reference/statements/exchange.md:
```diff
@@ -0,0 +1,42 @@
+# EXCHANGE Statement {#exchange}
+
+Exchanges the names of two tables or dictionaries atomically.
+This task can also be accomplished with a [RENAME](./rename.md) query using a temporary name. But the operation in not atomic in that case.
```
```suggestion
This task can also be accomplished with a [RENAME](./rename.md) query using a temporary name, but the operation is not atomic in that case.
```

olgarev commented on docs/en/sql-reference/statements/rename.md:
```diff
@@ -5,18 +5,57 @@ toc_title: RENAME
 
-## RENAME DATABASE {#misc_operations-rename_database}
-Renames database, it is supported only for Atomic database engine.
+Renames databases, tables or dictionaries. Several entities can be renamed in a single query.
```
```suggestion
Renames databases, tables, or dictionaries. Several entities can be renamed in a single query.
```

olgarev commented on docs/en/sql-reference/statements/rename.md:
```diff
@@ -5,18 +5,57 @@ toc_title: RENAME
-## RENAME DATABASE {#misc_operations-rename_database}
-Renames database, it is supported only for Atomic database engine.
+Renames databases, tables or dictionaries. Several entities can be renamed in a single query.
+Note that the `RENAME` query with several entities is non-atomic operation. To swap entity names atomically, use [EXCHANGE](./exchange.md) statement.
```
```suggestion
Note that the `RENAME` query with several entities is non-atomic operation. To swap entities names atomically, use [EXCHANGE](./exchange.md) statement.
```

olgarev commented on docs/en/sql-reference/statements/attach.md:
```diff
@@ -22,10 +29,12 @@ This query is used when starting the server. The server stores table metadata as
-### Сreate New Table And Attach Data {#create-new-table-and-attach-data}
+## Create New Table And Attach Data {#create-new-table-and-attach-data}
 
 **With specify path to table data**
```
```suggestion
### With specify path to table data {#attach-with-specified-path}
```

lehasm committed:
Update docs/en/sql-reference/statements/detach.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>

lehasm committed:
Update docs/en/sql-reference/statements/detach.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>

lehasm committed:
Update docs/en/sql-reference/statements/exchange.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>

lehasm committed:
Update docs/en/sql-reference/statements/rename.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>

lehasm committed:
Update docs/en/sql-reference/statements/rename.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>

lehasm committed:
Update docs/en/sql-reference/statements/attach.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>

lehasm committed:
Yet another header.
Other changes.

lehasm committed:
Translation to Russian

lehasm committed:
Atomic engine updates
Links added
Minor fixes.

Added ru settings
  default_replica_path
  default_replica_name

lehasm committed:
ru updated

olgarev commented on docs/en/engines/database-engines/atomic.md:
```diff
@@ -25,16 +25,16 @@ CREATE TABLE name UUID '28f1c61c-2970-457a-bffe-454156ddcfef' (n UInt64) ENGINE
 
 ### DROP/DETACH TABLE {#drop-detach-table}
 
 On `DROP TABLE` no data is removed, database `Atomic` just marks table as dropped by moving metadata to `/clickhouse_path/metadata_dropped/` and notifies background thread. Delay before final table data deletion is specify by [database_atomic_delay_before_drop_table_sec](../../operations/server-configuration-parameters/settings.md#database_atomic_delay_before_drop_table_sec) setting.
```
```suggestion
On `DROP TABLE` no data is removed, database `Atomic` just marks table as dropped by moving metadata to `/clickhouse_path/metadata_dropped/` and notifies background thread. Delay before final table data deletion is specified by the [database_atomic_delay_before_drop_table_sec](../../operations/server-configuration-parameters/settings.md#database_atomic_delay_before_drop_table_sec) setting.
```

olgarev commented on docs/ru/engines/database-engines/atomic.md:
```diff
@@ -25,16 +25,16 @@ CREATE TABLE name UUID '28f1c61c-2970-457a-bffe-454156ddcfef' (n UInt64) ENGINE
 ### RENAME TABLE {#rename-table}
 
-Запросы `RENAME` выполняются без изменения UUID и перемещения табличных данных. Эти запросы не ожидают завершения использующих таблицу запросов и будут выполнены мгновенно.
+Запросы [RENAME](../../sql-reference/statements/rename.md) выполняются без изменения UUID и перемещения табличных данных. Эти запросы не ожидают завершения использующих таблицу запросов и будут выполнены мгновенно.
```
```suggestion
Запросы [RENAME](../../sql-reference/statements/rename.md) выполняются без изменения UUID и перемещения табличных данных. Эти запросы не ожидают завершения использующих таблицу запросов и выполняются мгновенно.
```

olgarev commented on docs/en/engines/database-engines/atomic.md:
```diff
@@ -25,16 +25,16 @@ CREATE TABLE name UUID '28f1c61c-2970-457a-bffe-454156ddcfef' (n UInt64) ENGINE
 ### RENAME TABLE {#rename-table}
 
-`RENAME` queries are performed without changing UUID and moving table data. These queries do not wait for the completion of queries using the table and will be executed instantly.
+[RENAME](../../sql-reference/statements/rename.md) queries are performed without changing UUID and moving table data. These queries do not wait for the completion of queries using the table and will be executed instantly.
```
```suggestion
[RENAME](../../sql-reference/statements/rename.md) queries are performed without changing UUID and moving table data. These queries do not wait for the completion of queries using the table and are executed instantly.
```

olgarev commented on docs/ru/sql-reference/statements/attach.md:
```diff
@@ -3,16 +3,23 @@ toc_priority: 40
 ---
 
-# ATTACH Statement {#attach}
+# Выражение ATTACH {#attach}
```
```suggestion
# ATTACH {#attach}
```

olgarev commented on docs/ru/sql-reference/statements/exchange.md:
```diff
@@ -0,0 +1,42 @@
+toc_title: EXCHANGE
+---
+
+# Выражение EXCHANGE {#exchange}
```
```suggestion
# EXCHANGE {#exchange}
```

olgarev commented on docs/ru/sql-reference/statements/rename.md:
```diff
@@ -3,20 +3,58 @@ toc_priority: 48
 ---
 
-# RENAME Statement {#misc_operations-rename}
+# Выражение RENAME {#misc_operations-rename}
```
```suggestion
# RENAME {#misc_operations-rename}
```

lehasm committed:
Update docs/en/engines/database-engines/atomic.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>

lehasm committed:
Update docs/ru/engines/database-engines/atomic.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>

lehasm committed:
Update docs/en/engines/database-engines/atomic.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>

lehasm committed:
Update docs/ru/sql-reference/statements/attach.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>

lehasm committed:
Update docs/ru/sql-reference/statements/exchange.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>

lehasm committed:
Update docs/ru/sql-reference/statements/rename.md

Co-authored-by: olgarev <56617294+olgarev@users.noreply.github.com>
